### PR TITLE
Improve mass notification processing

### PR DESCRIPTION
**Pull request by nickvergessen**
### Steps
1. Create 70 users:
```sh
i=10
while [ $i -ne 70 ]
do
        i=$(($i+1))
        echo "$i"

	USERID=$(uuidgen)
	OC_PASS="123456" occ -vvv user:add --password-from-env --display-name "Dummy #$i" -g "dummies" $USERID
	occ -vvv user:setting $USERID settings email "test$i@yourdomain.tld"

done
```
2. Create a conversation as admin with group "dummies"
3. Start a call
4. Measure time around the section in talk:
https://github.com/nextcloud/spreed/blob/1aaf10e2653e764ed54a6abc9eac750753263a3c/lib/Notification/Listener.php#L234-L253
```php
			// Add this after https://github.com/nextcloud/spreed/blob/1aaf10e2653e764ed54a6abc9eac750753263a3c/lib/Notification/Listener.php#L233
			$start = microtime(true);
```
```php
			// Add this before https://github.com/nextcloud/spreed/blob/1aaf10e2653e764ed54a6abc9eac750753263a3c/lib/Notification/Listener.php#L255
			$end = microtime(true);
			\OC::$server->getLogger()->error('notify: ' . ($end - $start));
```
5. Measure the patch again with the beginTransaction and commit being commented out

With patch | Without patch
---|---
~0.01s | ~0.5s

### CODE REVIEW

File: lib/Handler.php [hunk 1]
```diff
@@ -96,23 +96,31 @@ public function delete(INotification $notification): array {
...
+			if ($shouldFlush) {
+				$this->manager->flush();
 			}
+		} catch (\Throwable $e) {
+			$this->connection->rollBack();
+			throw $e;
 		}
+		$this->connection->commit();
 
 		return $deleted;
```
Comment (come-nc) [inline lib/Handler.php:125]: $deleted will always be empty here. If intended replace by return []; and maybe add a comment.

Comment (nickvergessen) [reply lib/Handler.php:125]: Right, thanks for the pointed. there is yet another loop of events going out and in again. let me see if we can optimize that

Comment (nickvergessen) [reply lib/Handler.php:133]: Thanks for the hint. Fixed

**Review by vitormattos — APPROVED** [review]
Sounds good!
I liked the usage of array_chunk

**Pull request merged**